src/components/cylc/tree/index.js

```diff
@@ -14,58 +14,548 @@
-    // throw new Error('You must provide valid data to populate the tree!')
-    // a stopped workflow is valid, but won't have anything that we can use
-    // to populate the tree, only workflow data and empty families
+ * This is a simplified version of lodash's function with the same
+ * name, but that respects natural order for numbers, i.e. [1, 2, 10].
+ * Not [1, 10, 2].
+ *
+ * @private
+ * @param array {Array<object>} - list of string values, or of objects with string values
+ * @param value {object} - a value to be inserted in the list, or an object wrapping the value (see iteratee)
+ * @param iteratee {SortedIndexByIteratee=} - an optional function used to return the value of the element of the list}
+ * @param comparator {SortedIndexByComparator=} - function used to compare the newValue with otherValues in the list
+ */
+function sortedIndexBy (array, value, iteratee, comparator) {
+  if (array.length === 0) {
+    return 0
+  }
+  // If given a function, use it. Otherwise, simply use identity function.
+  const iterateeFunction = iteratee || ((value) => value)
+  // If given a function, use it. Otherwise, simply use locale sort with numeric enabled
+  const comparatorFunction = comparator || ((leftObject, leftValue, rightObject, rightValue) => DEFAULT_COMPARATOR(leftValue, rightValue) > 0)
+  let low = 0
+  let high = array.length
+
+  const newValue = iterateeFunction(value)
+
+  while (low < high) {
+    const mid = Math.floor((low + high) / 2)
+    const midValue = iterateeFunction(array[mid])
+    const higher = comparatorFunction(value, newValue, array[mid], midValue)
+    if (higher) {
+      low = mid + 1
+    } else {
+      high = mid
+    }
+  }
+  return high
+}
+
+/**
+ * @param {Workflow} workflow
+ */
+function clear (workflow) {
+  ['tree', 'lookup'].forEach(each => {
+    Object.keys(workflow[each]).forEach(key => {
+      Vue.delete(workflow[each], key)
+    })
+  })
+}
+
+/**
+ * @param {Workflow} workflow
+ */
+function isEmpty (workflow) {
+  return Object.keys(workflow.lookup).length === 0
+}
+
+/**
+ * @private
+ * @param {TreeNode} node
+ * @param {Lookup} lookup
+ */
+function recursivelyRemoveNode (node, lookup) {
+  const stack = [node]
+  while (stack.length > 0) {
+    const n = stack.pop()
+    Vue.delete(lookup, n.id)
+    if (n.children && n.children.length > 0) {
+      stack.push(...n.children)
+    }
+  }
+}
+
+// --- Workflow
+
+function addWorkflow (workflowNode, workflow, options) {
+  if (!workflow.lookup[workflowNode.id]) {
+    mergeWith(workflow.tree, workflowNode, mergeWithCustomizer)
+    Vue.set(workflow.lookup, workflowNode.id, workflow.tree)
+  }
+}
+
+// --- Cycle points
+
+/**
+ * @param {Workflow} workflow
+ * @param {CyclePointNode} cyclePoint
+ * @param {*} options
+ */
+function addCyclePoint (cyclePoint, workflow, options) {
+  const cyclePointsOrderDesc = options.cyclePointsOrderDesc !== undefined
+    ? options.cyclePointsOrderDesc
+    : DEFAULT_CYCLE_POINTS_ORDER_DESC
```

Comment:
>Can move inside the if block below.

Good catch! done!

>Should this be applied at render rather than at tree build as I'm not sure how stable this will be if the user changes the preference?

Good question. At the moment it's implemented in a way where users have to visit the settings page, and reload the page with a tree view (if already open). Created an issue to track that: https://github.com/cylc/cylc-ui/issues/694